controllers/selfnoderemediation_controller.go

```diff
@@ -154,13 +154,20 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 		return ctrl.Result{}, err
 	}
 
-	snrOrg := snr.DeepCopy()
 	defer func() {
-		if patchErr := r.patchSnrStatus(ctx, snr, snrOrg); patchErr != nil {
-			if returnErr == nil {
-				returnErr = patchErr
+		if updateErr := r.updateSnrStatus(ctx, snr); updateErr != nil {
+			if apiErrors.IsConflict(updateErr) {
+				if returnErr == nil {
+					if !returnResult.Requeue {
+						returnResult = ctrl.Result{RequeueAfter: time.Second}
+					}
+				}
 			} else {
-				returnErr = utilerrors.NewAggregate([]error{patchErr, returnErr})
+				if returnErr == nil {
```

Comment:
IIUC, looking at `utilerrors.NewAggregate` it seems that it can handle a `nil` item in the list, and if that so, we can always call it reducing the if/else nesting.